backend/src/main/java/com/github/backend/service/UserService.java

```diff
@@ -31,4 +34,19 @@ public User createUser(OAuth2User user, UserDto newUserDto){
                 null);
         return userRepo.save(createdUser);
     }
+
+    public void changeFavorites(Boulder boulder, OAuth2User user) {
+        User currentUser = getUserById(user.getAttributes().get("id").toString()).orElseThrow();
+        Optional<Boulder> favoriteBoulder= currentUser.getMyFavorites().stream().filter(favorite -> favorite.getId().equals(boulder.getId())).findFirst();
```

Comment:
same here :-) you can then get rid of that if statement ;-)

Reply:
I need the Optional here. Its possible that I haven't added this boulder to my favorites yet. In this case I want to add this boulder to my list, otherwise I want to delete it.